jina/peapods/pods/compoundpod.py

```diff
@@ -237,11 +236,15 @@ def rolling_update(self):
         """
         Update all pods of this compound pod.
         """
-        for i in range(len(self.replicas)):
-            replica = self.replicas[i]
-            replica.close()
-            _args = self.replicas_args[i]
-            _args.noblock_on_start = False
-            new_replica = Pod(_args)
-            self.enter_context(new_replica)
-            self.replicas[i] = new_replica
+        try:
+            for i in range(len(self.replicas)):
+                replica = self.replicas[i]
+                replica.deactivate()
+                replica.close()
```

Comment:
As mentioned in my other comment, I believe this should be one command towards the replica and the replica itself cares about the deactivate/close order of its Peas. The CompoundPod should not need to orchestrate these calls.

Reply:
Makes sense, let me try to handle in another iteration. U can create a ticket from this comment and I will take care of it